src/main/java/com/google/devtools/build/lib/analysis/actions/StrippingPathMapper.java

```diff
@@ -0,0 +1,268 @@
+      }
+
+      @Override
+      public List<String> mapCustomStarlarkArgs(List<String> args) {
+        if (!isStarlarkAction) {
+          return args;
+        }
+        // Add your favorite Starlark mnemonic that needs custom arg processing here.
+        if (!mnemonic.contains("Android")
+            && !mnemonic.equals("MergeManifests")
+            && !mnemonic.equals("StarlarkRClassGenerator")) {
+          return args;
+        }
+        // Add your favorite arg to custom-process here. When Bazel finds one of these in the
+        // argument list (an argument name), it strips output path prefixes from the following
+        // argument (the argument value).
+        ImmutableList<String> starlarkArgsToStrip =
+            ImmutableList.of(
+                "--primaryData",
+                "--directData",
+                "--data",
+                "--resources",
+                "--mergeeManifests",
+                "--library");
+        for (int i = 1; i < args.size(); i++) {
+          if (starlarkArgsToStrip.contains(args.get(i - 1))) {
+            args.set(i, argStripper.strip(args.get(i)));
+          }
+        }
+        return args;
+      }
+
+      @Override
+      public ExceptionlessMapFn<Object> getMapFn(@Nullable String previousFlag) {
+        if (isJavaAction && "--javacopts".equals(previousFlag)) {
+          return structuredArgStripper;
+        }
+        return MapFn.DEFAULT;
+      }
+    };
+  }
+
+  /**
+   * Utility class to strip output path configuration prefixes from arbitrary strings.
+   */
+  private static class StringStripper {
+    private final Pattern pattern;
+    private final String outputRoot;
+
+    public StringStripper(String outputRoot) {
+      this.outputRoot = outputRoot;
+      this.pattern = stripPathsPattern(outputRoot);
+    }
+
+    /**
+     * Returns the regex to strip output paths from a string.
+     *
+     * <p>Supports strings with multiple output paths in arbitrary places. For example
+     * "/path/to/compiler bazel-out/x86-fastbuild/foo src/my.src -Dbazel-out/arm-opt/bar".
+     *
+     * <p>Doesn't strip paths that would be non-existent without config prefixes. For example,
+     * these are unchanged: "bazel-out/x86-fastbuild", "bazel-out;foo", "/path/to/compiler
+     * bazel-out".
+     *
+     * @param outputRoot root segment of output paths (e.g. "bazel-out")
+     */
+    private static Pattern stripPathsPattern(String outputRoot) {
+      // Match "bazel-out" followed by a slash followed by any combination of word characters, "_",
+      // and "-", followed by another slash. This would miss substrings like
+      // "bazel-out/k8-fastbuild". But those don't represent actual outputs (all outputs would have
+      // to have names beneath that path). So we're not trying to replace those.
+      return Pattern.compile(outputRoot + "/[\\w_-]+/");
+    }
+
+    public String strip(String str) {
+      return pattern.matcher(str).replaceAll(outputRoot + "/");
```

Comment:
I modified this to 

```
private static PathFragment strip(PathFragment execPath) {
    String config = "/" + execPath.subFragment(1, 2).getPathString();
    String mappedPath = execPath.getPathString().replace(config, "");
    return PathFragment.create(mappedPath);
 }
```

to take care of paths in the form `blaze-out/k8-fasbuilds/bin/foo/blaze-out/k8-fasbuilds/genfiles/bar`. The existing one will make the path look like `blaze-out/bin/foo/blaze-out/k8-fasbuilds/genfiles/bar`